ap/classnotes/utils.py

```diff
@@ -97,7 +97,7 @@ def get_leaveslip(trainee, roll):
   individualslips = IndividualSlip.objects.filter(trainee=trainee, status='A', rolls__in=[roll])
   roll_start_datetime = datetime.combine(roll.date, roll.event.start)
   roll_end_datetime = datetime.combine(roll.date, roll.event.end)
-  groupslips = GroupSlip.objects.filter(trainee=trainee, status='A', start__lte=roll_start_datetime, end__gte=roll_end_datetime)
+  groupslips = GroupSlip.objects.filter(trainees=trainee, status='A', start__lte=roll_start_datetime, end__gte=roll_start_datetime)
```

Comment:
I don't know if it catches every case. How about...
```suggestion
  groupslips = GroupSlip.objects.filter(trainees=trainee, status='A', start__lte=roll_end_datetime, end__gte=roll_start_datetime)
```
See https://github.com/attendanceproject/djattendance/blob/99884f39a1486515692d4cb3d1d8367564b428af/ap/accounts/models.py#L388

Reply:
Yeah that looks good